### ISSUE TITLE
Node JsxExpression did not pass test 'isExpression'

### ISSUE THREAD

**Issue by koxu1996** [KonstantinSimeonov/tsx-control-statements#232]
Let us consider the following component:
```ts
import React from "react";
import { If } from "./utils/tsx-control-statements/components";

export default const Foo = () => {
  const showContent = Math.random() > 0.5;
  const content = <If condition={showContent}>Foo</If>;
  return <div>{content}</div>;
};
```
It works with `yarn run react-scripts build`, but fails for `yarn run react-scripts start`:

```
Compiled with problems:

ERROR in ./foo.tsx

Module build failed (from ./node_modules/ts-loader/index.js):
Error: Debug Failure. False expression: Unexpected node.
Verbose Debug Information: Node JsxExpression did not pass test 'isExpression'.
    at visitNode (/app/node_modules/typescript/lib/typescript.js:91379:18)
    at visitEachChildOfVariableDeclaration (/app/node_modules/typescript/lib/typescript.js:91894:240)
    at Object.visitEachChild (/app/node_modules/typescript/lib/typescript.js:91597:42)
    at visitNodes (/app/src/utils/tsx-control-statements/transformer.ts:36:13)
    at /app/src/utils/tsx-control-statements/transformer.ts:38:20
    at visitArrayWorker (/app/node_modules/typescript/lib/typescript.js:91453:48)
    at visitNodes (/app/node_modules/typescript/lib/typescript.js:91417:23)
    at visitEachChildOfVariableDeclarationList (/app/node_modules/typescript/lib/typescript.js:91897:72)
    at Object.visitEachChild (/app/node_modules/typescript/lib/typescript.js:91597:42)
    at visitNodes (/app/src/utils/tsx-control-statements/transformer.ts:36:13)
```

---

I found the temporary workaround:

```patch
-  const content = <If condition={showContent}>Foo</If>;
+  const content = <><If condition={showContent}>Foo</If></>;
```

**Comment by koxu1996** [comment]
I also modified TypeScript's `visitNode()` to see how **failing node** looks like:

<details>
  <summary>original</summary>
  
  ```
  <ref *2> NodeObject {
    pos: 173,
    end: 210,
    flags: 0,
    modifierFlagsCache: 0,
    transformFlags: 2,
    parent: <ref *1> NodeObject {
      pos: 163,
      end: 210,
      flags: 0,
      modifierFlagsCache: 536870912,
      transformFlags: 2,
      parent: NodeObject {
        pos: 155,
        end: 210,
        flags: 2,
        modifierFlagsCache: 536870912,
        transformFlags: 4457474,
        parent: [NodeObject],
        kind: 258,
        declarations: [Array]
      },
      kind: 257,
      symbol: SymbolObject {
        flags: 2,
        escapedName: 'content',
        declarations: [Array],
        exports: Map(0) {},
        valueDeclaration: [Circular *1],
        parent: undefined,
        id: 1457,
        isReferenced: 1160127
      },
      localSymbol: undefined,
      locals: undefined,
      nextContainer: undefined,
      name: IdentifierObject {
        pos: 163,
        end: 171,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: [Circular *1],
        kind: 79,
        originalKeywordKind: undefined,
        escapedText: 'content',
        flowNode: [Object]
      },
      initializer: [Circular *2],
      type: undefined,
      exclamationToken: undefined
    },
    kind: 281,
    openingElement: <ref *3> NodeObject {
      pos: 173,
      end: 202,
      flags: 0,
      modifierFlagsCache: 0,
      transformFlags: 2,
      parent: [Circular *2],
      kind: 283,
      tagName: IdentifierObject {
        pos: 175,
        end: 177,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: [Circular *3],
        kind: 79,
        originalKeywordKind: undefined,
        escapedText: 'If',
        flowNode: [Object],
        id: 5350
      },
      typeArguments: undefined,
      attributes: NodeObject {
        pos: 177,
        end: 201,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 2,
        parent: [Circular *3],
        kind: 289,
        properties: [Array],
        symbol: [SymbolObject],
        nextContainer: [NodeObject],
        _children: [Array]
      },
      id: 5349,
      _children: [ [TokenObject], [IdentifierObject], [NodeObject], [TokenObject] ]
    },
    children: [
      NodeObject {
        pos: 202,
        end: 205,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 2,
        parent: [Circular *2],
        kind: 11,
        text: 'Foo',
        containsOnlyTriviaWhiteSpaces: false
      },
      pos: 202,
      end: 205,
      hasTrailingComma: false,
      transformFlags: 2
    ],
    closingElement: <ref *4> NodeObject {
      pos: 205,
      end: 210,
      flags: 0,
      modifierFlagsCache: 0,
      transformFlags: 2,
      parent: [Circular *2],
      kind: 284,
      tagName: IdentifierObject {
        pos: 207,
        end: 209,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: [Circular *4],
        kind: 79,
        originalKeywordKind: undefined,
        escapedText: 'If',
        flowNode: [Object],
        id: 5352
      }
    },
    id: 5345,
    _children: [
      <ref *3> NodeObject {
        pos: 173,
        end: 202,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 2,
        parent: [Circular *2],
        kind: 283,
        tagName: [IdentifierObject],
        typeArguments: undefined,
        attributes: [NodeObject],
        id: 5349,
        _children: [Array]
      },
      NodeObject {
        pos: 202,
        end: 205,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: [Circular *2],
        kind: 351,
        _children: [Array]
      },
      <ref *4> NodeObject {
        pos: 205,
        end: 210,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 2,
        parent: [Circular *2],
        kind: 284,
        tagName: [IdentifierObject]
      }
    ]
  }
  ```
</details>

<details>
  <summary>after tsx-control-statement transformation</summary>
  
  ```
  NodeObject {
    pos: -1,
    end: -1,
    flags: 8,
    modifierFlagsCache: 0,
    transformFlags: 2,
    parent: undefined,
    kind: 291,
    dotDotDotToken: undefined,
    expression: NodeObject {
      pos: -1,
      end: -1,
      flags: 8,
      modifierFlagsCache: 0,
      transformFlags: 2,
      parent: undefined,
      kind: 224,
      condition: NodeObject {
        pos: -1,
        end: -1,
        flags: 8,
        modifierFlagsCache: 0,
        transformFlags: 2,
        parent: undefined,
        kind: 214,
        expression: [NodeObject]
      },
      questionToken: TokenObject {
        pos: -1,
        end: -1,
        flags: 8,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: undefined,
        kind: 57
      },
      whenTrue: TokenObject {
        pos: -1,
        end: -1,
        flags: 8,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: undefined,
        kind: 10,
        text: 'Foo',
        singleQuote: undefined,
        hasExtendedUnicodeEscape: undefined
      },
      colonToken: TokenObject {
        pos: -1,
        end: -1,
        flags: 8,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: undefined,
        kind: 58
      },
      whenFalse: TokenObject {
        pos: -1,
        end: -1,
        flags: 8,
        modifierFlagsCache: 0,
        transformFlags: 0,
        parent: undefined,
        kind: 104
      }
    }
  }
  ```
Notice that root node's **kind** was changed from `JsxElement` (281) to  `JsxExpression` (291) - this seems to be wrong, but I have no idea about AST of TypeScript.

**Comment by KonstantinSimeonov** [comment]
@koxu1996 sorry for taking an eternity to work on this - should be fixed in `5.0` I just released, just keep in mind it requires ts `4.9` now.
